### ISSUE TITLE
Allow API to specify request and response headers for an operation

### ISSUE THREAD

**Issue by gheine** [apicollective/apibuilder#519]
Some operations may require a request header (e.g. for authorization or session tracking) or take optional headers. Currently there is no way to express which request headers (and response headers likewise) are supported/required/offered by a resource's operation. 

Simple way to implement this would be to add an additional parameter_location enum value 'Header' and allow operations to take parameters of this type, which will place the param on the request header.

More involved (but possibly cleaner) solution would be to add optional request_header and response_header fields to operation, which can be used to reference entries in the service's headers, like: 

```json
"header_ref": { "fields": [
  { "name": "name", "type": "string", "description": "The name of a header referenced in the service's headers section" },
  { "name": "required", "type": "boolean", "default": "false", "description": "Denotes whether or not this header is required in the request, or whether or not it is always present in the response" }
]}
```

**Comment by mbryzek** [comment]
makes sense. only open question before we can add support is to figure out how authorization headers should be supported as that may impact the format of per operation headers (e.g. can we avoid repeating the exact same headers on every operation)

**Comment by gheine** [comment]
Not sure what you mean. If every operation takes an authorization header, I think it should be documented as such and every operation should list that header.

**Comment by gheine** [comment]
Here's what I think this could look like: https://github.com/mbryzek/apidoc/pull/524 (PR #520 was the simple/incomplete solution)

**Comment by gheine** [comment]
Fixed in #520
